File: in_Java/0645 Set Mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        Map<Integer,Integer> hm = new HashMap<>();
        for(int i : nums){
            if(!hm.containsKey(i)) hm.put(i,1);
            else hm.put(i, hm.get(i)+1);
        }
        int[] res = new int[2];
        for(int i = 1; i <= nums.length; i++){
            if(!hm.containsKey(i)) res[1] = i;
            else{
                if(hm.get(i) == 2) res[0] = i;
            }
        }
        return res;
    }
}

File: in_Python/0645 Set Mismatch.py
class Solution(object):
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        n = len(nums)+1
        d = {}
        res = [0,0]
        for i in nums:
            if i not in d:
                d[i] = 1
            else:
                res[0] = i
                d[i] = 1
        for i in range(1,n):
            if i not in d:
                res[1] = i
                break
        return res